packages/neuron-wallet/src/controllers/api.ts

```diff
@@ -620,8 +620,19 @@ export default class ApiController {
           err.code = NODE_DISCONNECTED_CODE
         }
 
-        if (err.message?.code !== undefined && !Number.isNaN(err.message.code)) {
-          err.code = err.message.code
+        try {
+          /**
+           * error.message from ckb node is a stringified error object with code and message
+           */
+          const e = JSON.parse(err.message)
```

Comment:
Is the rpc result `object`?

Reply:
Yes, `error.message` consists of `{ code: number, message: string }` but is stringified.